internal/chain/create_key_book.go

```diff
@@ -14,77 +14,63 @@ type CreateKeyBook struct{}
 
 func (CreateKeyBook) Type() types.TxType { return types.TxCreateKeyBook }
 
-func checkCreateKeyBook(st *StateManager, tx *transactions.GenTransaction) ([]*protocol.SigSpec, *url.URL, error) {
+func (CreateKeyBook) Validate(st *StateManager, tx *transactions.GenTransaction) error {
 	if _, ok := st.Sponsor.(*state.AdiState); !ok {
-		return nil, nil, fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
+		return fmt.Errorf("invalid sponsor: want %v, got %v", types.ChainTypeAdi, st.Sponsor.Header().Type)
 	}
 
 	body := new(protocol.CreateSigSpecGroup)
 	err := tx.As(body)
 	if err != nil {
-		return nil, nil, fmt.Errorf("invalid payload: %v", err)
+		return fmt.Errorf("invalid payload: %v", err)
 	}
 
 	if len(body.SigSpecs) == 0 {
-		return nil, nil, fmt.Errorf("cannot create empty sig spec group")
+		return fmt.Errorf("cannot create empty sig spec group")
 	}
 
 	sgUrl, err := url.Parse(body.Url)
 	if err != nil {
-		return nil, nil, fmt.Errorf("invalid target URL: %v", err)
+		return fmt.Errorf("invalid target URL: %v", err)
 	}
 
 	if !sgUrl.Identity().Equal(st.SponsorUrl) {
-		return nil, nil, fmt.Errorf("%q does not belong to %q", sgUrl, st.SponsorUrl)
+		return fmt.Errorf("%q does not belong to %q", sgUrl, st.SponsorUrl)
 	}
 
 	entries := make([]*protocol.SigSpec, len(body.SigSpecs))
 	for i, chainId := range body.SigSpecs {
 		entry := new(protocol.SigSpec)
 		err = st.LoadAs(chainId, entry)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to fetch sig spec: %v", err)
+			return fmt.Errorf("failed to fetch sig spec: %v", err)
```

Comment:
"Failed to fetch Key Page"

Reply:
Yes, but I want to do that as part of [AC-528](https://accumulate.atlassian.net/browse/AC-528)